baseplate/frameworks/queue_consumer/kombu.py

```diff
@@ -94,23 +101,29 @@ def handle(self, message: kombu.Message) -> None:
             # handle the error (publish it to error reporting)
             with self.baseplate.make_server_span(context, self.name) as span:
                 delivery_info = message.delivery_info
+                message_body = None
+                message_body = message.decode()
                 span.set_tag("kind", "consumer")
                 span.set_tag("amqp.routing_key", delivery_info.get("routing_key", ""))
                 span.set_tag("amqp.consumer_tag", delivery_info.get("consumer_tag", ""))
                 span.set_tag("amqp.delivery_tag", delivery_info.get("delivery_tag", ""))
                 span.set_tag("amqp.exchange", delivery_info.get("exchange", ""))
-                self.handler_fn(context, message.decode(), message)
+                self.handler_fn(context, message_body, message)
         except Exception as exc:
             logger.exception(
                 "Unhandled error while trying to process a message.  The message "
                 "has been returned to the queue broker."
             )
-            message.requeue()
+            if self.error_handler_fn:
+                self.error_handler_fn(context, message_body, message)
+            else:
+                message.requeue()
             if isinstance(exc, FatalMessageHandlerError):
                 logger.info("Recieved a fatal error, terminating the server.")
                 raise
         else:
-            message.ack()
+            if not self.error_handler_fn:
+                message.ack()
```

Comment:
I don't think we want to do this, we should just `ack` if there's no error, I don't think that adding an `error_handler_fn` should affect `handler_fn`.

Reply:
The problem with `always` calling `ack` in a finally clause is that if we call `reject` in the `error_handler_fn`, it doesn't allow you to call `ack` on the message again.

Reply:
I guess what you're saying is to remove the `ack` from the `else` clause and move it in the `try` clause

Reply:
We wouldn't call `ack` in that case because this code only runs if an exception isn't raised (it's an `else`, not a `finally`), we want this outside of the try block since we don't want to catch `ack` errors, the try block is just for errors during message handling.